src/dataset-builder/DatasetBuilder.ts

```diff
@@ -659,7 +661,7 @@ export const DatasetBuilderContents = ({
       requestValid &&
         h(ActionBar, {
           prompt: h(Fragment, [
-            datasetRequestParticipantCount.status === 'Ready'
+            datasetRequestParticipantCount.status === 'Ready' && datasetRequestParticipantCount.state?.result.total
```

Comment:
I have _no_ clue why this is suddenly required. All of a sudden tests failed without this. However, mocking out the method didn't help, explicitly mocking error handling didn't help, and it seemed to pass if I ran a single test but not if I ran multiple. I am lost, so if anyone has any better ideas I would love to hear them. Otherwise I guess we add a few explicit null checks.

Reply:
When I log `datasetRequestParticipantCount`, I see some instances of `{ status: 'Ready', state: undefined }`.

It looks like, in some tests, no mock return value / implementation is set up for `DataRepo().dataset(...).getSnapshotBuilderCount()` and so it returns undefined, which then causes an error when trying to access `datasetRequestParticipantCount.state.result.total`.